src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    private String filePath;
+    private String dirPath;
+
+    /**
+     * @param filePath The file path of the file which is used to store task list.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        parseFilePath();
+    }
+
+    /**
+     * Extracts directory path from the file path.
+     */
+    public void parseFilePath() {
+        String pattern = "(.*)/(.*)$";
+        Pattern r = Pattern.compile(pattern);
+        Matcher m = r.matcher(filePath);
+        if (m.find()) {
+            this.dirPath = m.group(1);
+        }
+    }
+
+    /**
+     * Loads task list from the file and returns the task list.
+     * @return The task list which is loaded from the file.
+     * @throws DukeException Exception if there is error when loading from the file.
+     */
+    public ArrayList<Task> load() throws DukeException {
```

Comment:
Can this method name be _loadTasksFromStorage_ to explain the method more intuitively?